staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go

```diff
@@ -18,6 +18,13 @@ package v1alpha1
 
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+const (
+	// InTreePluginMechanism means components should use the in-tree plugin for volume operations
+	InTreePluginMechanism string = "in-tree"
+	// CSIPluginMechanism means components should use the CSI Driver for volume operations
+	CSIPluginMechanism string = "csi"
```

Comment:
Define a custom `type PluginMechanism string` rather than using string for these.